data-prepper-plugins/common/src/main/java/com/amazon/dataprepper/plugins/buffer/BlockingBuffer.java

```diff
@@ -41,8 +45,15 @@
     private static final String ATTRIBUTE_BATCH_SIZE = "batch_size";
 
     private final int batchSize;
+    private final int bufferCapacity;
     private final BlockingQueue<T> blockingQueue;
     private final String pipelineName;
+    private final AtomicInteger numInflightRecords = new AtomicInteger();
```

Comment:
These are not exactly "inflight" records. These are uncheckpointed records that have been written. In-flight would refer to records that have been read but not yet checkpointed. My understanding was that we did not want the in-flight to be greater than buffer capacity. But it seems that what we have here is in-flight + unread cannot be greater than buffer capacity. Just want to clarify that this is what we want.

Reply:
Correct, we'll have to make it clear that a buffer's capacity is only freed up once an item has been both _read and 
 acknowledged_. (Present behavior is only after an item is _read_). 

We can revisit later if we think having granular control over both internal queue capacity and in-flight capacity is necessary, but for now combining the capacities keeps things simple.